sse2neon.h

```diff
@@ -9059,10 +9118,62 @@ FORCE_INLINE int _mm_cmpestrz(__m128i a,
                               int lb,
                               const int imm8)
 {
-    SSE2NEON_CMPESTRX_SET_UPPER(bound, imm8);
+    SSE2NEON_CMPSTR_SET_UPPER(bound, imm8);
     return lb <= (bound - 1);
 }
 
+#define SSE2NEON_CMPISTRX_GET_LEN(a, b, la, lb)                                \
```

Comment:
I am thinking of the possibility to shorten this macro by using another helper. Can you refine it?

Reply:
@jserv you mean for shortening the content of this macro (i.e., calculation of `la` and `lb`) or a helper like:
```c
#define SSE2NEON_GET_LEN(..., IM) \
  #define SSE2NEON_##IM##_GET_LEN(...)
```

Reply:
Sort of. Macros should be defined as shorter as possible. We can aggregate the existing macros instead of defining the opposite.

Reply:
Done.
As the difference of `PCMPESTR*` and `PCMPISTR*` just the way to calculate the length of string, I apply the following strategy for implementation:
```c
#define SSE2NEON_COMP_AND_AGG(a, b, la, lb, imm8, IE) \
    SSE2NEON_CMPSTR_SET_UPPER(bound, imm8);                        \
    SSE2NEON_##IE##_GET_LEN(a, b, la, lb);                         \
    int r2 = (_sse2neon_cmpfunc_table[imm8 & 0x0f])(a, la, b, lb); \
    r2 = _sse2neon_sido_negative(r2, lb, imm8, bound)
```